### PR TITLE
DTSPB-4270 Long term fix Data Migration cases

### PR DESCRIPTION
**Pull request by IswaryaPepakayala**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DTSPB-4270

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/probate/transformer/CallbackResponseTransformer.java [hunk 1]
```diff
@@ -1823,4 +1833,12 @@ private List<CollectionMember<HandoffReason>> getHandoffReasonList(
         }
         return collectionMembers;
     }
+
+    private LocalDateTime getLastModifiedDate(String eventId, LocalDateTime lastModifiedDateForDormant) {
+        boolean noneMatch = EXCLUDED_EVENT_LIST.stream().noneMatch(s -> s.equals(eventId));
+        if (noneMatch) {
```
Comment (FeliTam) [inline src/main/java/uk/gov/hmcts/probate/transformer/CallbackResponseTransformer.java]: can we rename this to shouldSetDate

File: src/test/java/uk/gov/hmcts/probate/transformer/CallbackResponseTransformerTest.java [hunk 2]
```diff
@@ -2705,12 +2705,13 @@ private static Stream<String> invalidValue() {
     @Test
     void shouldTransformHandoffReason() {
```
Comment (FeliTam) [inline src/test/java/uk/gov/hmcts/probate/transformer/CallbackResponseTransformerTest.java]: please rename the test

File: src/test/java/uk/gov/hmcts/probate/transformer/CallbackResponseTransformerTest.java [hunk 3]
```diff
@@ -4274,6 +4275,17 @@ void shouldReturnNullWhenHandOffSiteIsNo() {
...
+    void shouldReturnDateWhenEventIsNotMatched() {
+        caseDataBuilder.applicationType(ApplicationType.PERSONAL);
+        when(callbackRequestMock.getEventId()).thenReturn("eventId");
+        when(callbackRequestMock.getCaseDetails()).thenReturn(caseDetailsMock);
+        when(caseDetailsMock.getData()).thenReturn(caseDataBuilder.build());
+
+        CallbackResponse callbackResponse = underTest.transform(callbackRequestMock);
+        assertNotNull(callbackResponse.getData().getLastModifiedDateForDormant());
+    }
+
```
Comment (FeliTam) [inline src/test/java/uk/gov/hmcts/probate/transformer/CallbackResponseTransformerTest.java:4443]: Can we also add a test for the History Correction event please?

File: src/test/java/uk/gov/hmcts/probate/transformer/CallbackResponseTransformerTest.java (hunk 2 above)
Comment (IswaryaPepakayala) [reply src/test/java/uk/gov/hmcts/probate/transformer/CallbackResponseTransformerTest.java]: Added

File: src/main/java/uk/gov/hmcts/probate/transformer/CallbackResponseTransformer.java (hunk 1 above)
Comment (IswaryaPepakayala) [reply src/main/java/uk/gov/hmcts/probate/transformer/CallbackResponseTransformer.java]: changed

**Review by FeliTam — DISMISSED** [review]

**Review by FeliTam — APPROVED** [review]

**Pull request merged**